packages/hash/datastore/postgres/migration/1648547815337_base.ts

```diff
@@ -0,0 +1,613 @@
+      },
+      path: {
+        type: "text",
+        notNull: true,
+        primaryKey: true,
+        comment:
+          "The JSON path of the aggregation on the source entity's properties JSON blob",
+      },
+      source_entity_version_ids: {
+        type: "uuid[]",
+        notNull: true,
+        primaryKey: true,
+        comment: stripNewLines(`
+          The entity version ids of the source entity's versions where
+          this aggregation exists.
+        `),
+      },
+      operation: {
+        type: "jsonb",
+        notNull: true,
+        comment: "The aggregation operation",
+      },
+      created_by_account_id: {
+        type: "uuid",
+        notNull: true,
+        comment: "The account that created this aggregation",
+      },
+      created_at: {
+        type: "timestamp with time zone",
+        notNull: true,
+        comment: "The time at which the first version of this type was created",
+      },
+    },
+    {
+      ifNotExists: true,
+      comment: "Stores aggregations of entities",
+    },
+  );
+
+  pgm.createTable(
+    "verification_codes",
+    {
+      verification_id: { type: "uuid", notNull: true },
+      account_id: { type: "uuid", notNull: true },
+      user_id: { type: "uuid", notNull: true },
+      verification_code: { type: "text", notNull: true },
+      email_address: { type: "text", notNull: true },
+      used: { type: "boolean", notNull: true, default: false },
+      number_of_attempts: { type: "integer", notNull: true, default: 0 },
+      created_at: { type: "timestamp with time zone", notNull: true },
+    },
+    {
+      ifNotExists: true,
+      comment:
+        "Stores verification codes used for passwordless authentication and email verification",
+    },
+  );
+
+  pgm.createTable(
+    "session",
+    {
+      sid: {
+        type: "text",
+        primaryKey: true,
+      },
+      sess: {
+        type: "jsonb",
+        notNull: true,
+      },
+      expire: {
+        type: "timestamp with time zone",
+        notNull: true,
+      },
+    },
+    {
+      ifNotExists: true,
+      comment: stripNewLines(`
+        \`connect-db-simple\` express session store (based on \`node_modules/connect-pg-simple/table.sql\`)
+
+        Note: column names cannot be modified, the table name can be modified but must also be passed
+        to \`connect-db-simple\` as a parameter in \`src/auth/session.ts\`
+      `),
+    },
+  );
+
+  pgm.createIndex("session", "expire", {
+    name: "session_expire",
+    ifNotExists: true,
+  });
+
+  pgm.createSchema("realtime", { ifNotExists: true });
+
+  pgm.createTable(
+    { schema: "realtime", name: "ownership" },
+    {
+      slot_name: { type: "text", primaryKey: true },
+      slot_owner: { type: "uuid", notNull: true },
+      ownership_expires_at: { type: "timestamp with time zone" },
+    },
+    { ifNotExists: true },
+  );
+}
+
+/** Rolling back would cause data loss, as tables would be destroyed. */
```

Comment:
I think it would be nice to add a notice in the README to explain why down is set to false if it is. Documenting that seems like it would be helpful

Reply:
Do you mean explaining in the README what it means if a `down` migration is set to `false`? In which case that's covered in step 2. of the "Creating a new migration" section.

If you mean that every time we set `down` to `false` we should add it to the README, I'm not sure I would agree. I think a comment in the specific migration file would suffice.

Reply:
I mean that we should expect a comment in case there is no down migration, which could be a guideline added to the README

Reply:
Ah, thanks for clarifying. I've added this in https://github.com/hashintel/hash/pull/438/commits/bdf602413a0a7bb1d69817a356cc998b18785c69